File: src/main/java/pl/allegro/experiments/chi/chiserver/infrastructure/HttpSessionConfig.java
package pl.allegro.experiments.chi.chiserver.infrastructure;

import org.springframework.context.annotation.Bean;
import org.springframework.session.data.mongo.JdkMongoSessionConverter;
import org.springframework.session.data.mongo.config.annotation.web.http.EnableMongoHttpSession;

import java.time.Duration;

@EnableMongoHttpSession
class HttpSessionConfig {

    @Bean
    JdkMongoSessionConverter jdkMongoSessionConverter() {
        return new JdkMongoSessionConverter(Duration.ofDays(1));
    }
}

File: src/main/kotlin/pl/allegro/experiments/chi/chiserver/infrastructure/HttpSessionConfig.kt
package pl.allegro.experiments.chi.chiserver.infrastructure

import org.springframework.context.annotation.Bean
import org.springframework.session.data.mongo.JdkMongoSessionConverter
import org.springframework.session.data.mongo.config.annotation.web.http.EnableMongoHttpSession

@EnableMongoHttpSession
class HttpSessionConfig {

    @Bean
    fun jdkMongoSessionConverter(): JdkMongoSessionConverter {
        return JdkMongoSessionConverter()
    }
}
